File: samples/snippets/csharp/VS_Snippets_VBCSharp/VBPowerPacksDataRepeaterDeletingItems/CS/DeletingItems.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Web.ClientServices
;

namespace DeletingItemsCS
{
    public partial class DeletingItems : Form
    {
        public DeletingItems()
        {
            InitializeComponent();
        }

        private void employeesBindingNavigatorSaveItem_Click(object sender, EventArgs e)
        {
            this.Validate();
            this.employeesBindingSource.EndEdit();
            this.tableAdapterManager.UpdateAll(this.northwndDataSet);

        }

        private void DeletingItems_Load(object sender, EventArgs e)
        {
            // TODO: This line of code loads data into the 'northwndDataSet.Employees' table. You can move, or remove it, as needed.
            this.employeesTableAdapter.Fill(this.northwndDataSet.Employees);

        }
        // <Snippet1>
        private void DataRepeater1_DeletingItems(object sender, 
            Microsoft.VisualBasic.PowerPacks.DataRepeaterAddRemoveItemsCancelEventArgs e)
        {
            // Check whether the user is a supervisor.

            ClientRolePrincipal rolePrincipal =
                System.Threading.Thread.CurrentPrincipal
                as ClientRolePrincipal;

            if (rolePrincipal.IsInRole("supervisor") == false)
            {
                e.Cancel = true;
                MessageBox.Show("You are not authorized to delete.");
            }
        }   
        // </Snippet1>
    }
}

File: samples/snippets/visualbasic/VS_Snippets_VBCSharp/VBPowerPacksDataRepeaterDeletingItems/VB/DeletingItems.vb
Public Class DeletingItems

    Private Sub EmployeesBindingNavigatorSaveItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles EmployeesBindingNavigatorSaveItem.Click
        Me.Validate()
        Me.EmployeesBindingSource.EndEdit()
        Me.TableAdapterManager.UpdateAll(Me.NorthwndDataSet)

    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        'TODO: This line of code loads data into the 'NorthwndDataSet.Employees' table. You can move, or remove it, as needed.
        Me.EmployeesTableAdapter.Fill(Me.NorthwndDataSet.Employees)

    End Sub
    ' <Snippet1>
    Private Sub DataRepeater1_DeletingItems(
        ByVal sender As Object, 
        ByVal e As Microsoft.VisualBasic.PowerPacks.DataRepeaterAddRemoveItemsCancelEventArgs
      ) Handles DataRepeater1.DeletingItems

        ' Check whether the user is a supervisor.
        If My.User.IsInRole("Supervisor") = False Then
            ' Cancel the deletion and display a message.
            e.Cancel = True
            MsgBox("You are not authorized to delete.")
        End If
    End Sub
    ' </Snippet1>
End Class
